opencog/query/PatternTerm.h

```diff
@@ -94,6 +94,9 @@ class PatternTerm
 	// literally.
 	int _quotation_level;
 
+	// True if the atom has been wrapped by a LocalQuoteLink
+	bool _local_quote;
```

Comment:
Don't introduce more complexity!  Just increment and decrement the quote level!   Keep it simple.